src/ko/agitoon/src/eu/kanade/tachiyomi/extension/ko/agitoon/AgiToon.kt

```diff
@@ -0,0 +1,224 @@
+        doc.select("script[src*=data/webtoon]").flatMap { scriptEl ->
+            var listIdx: Int
+            client.newCall(GET(scriptEl.absUrl("src"), headers))
+                .execute().body.string()
+                .also {
+                    listIdx = it.substringBefore(" = ")
+                        .substringAfter("data")
+                        .toInt()
+                }
+                .substringAfter(" = ")
+                .removeSuffix(";")
+                .let { json.decodeFromString<List<SeriesItem>>(it) }
+                .onEach { it.listIndex = listIdx }
+        }
+    }
+
+    private fun List<SeriesItem>.getPageChunk(page: Int): MangasPage {
+        return MangasPage(
+            mangas = subList((page - 1) * 24, min(page * 24, size))
+                .map { it.toSManga(cdnUrl) },
+            hasNextPage = (page + 1) * 24 <= size,
+        )
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        return Observable.just(
+            db.sortedByDescending { it.hot }.getPageChunk(page),
+        )
+    }
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> {
+        return Observable.just(
+            db.sortedByDescending { it.updatedAt }.getPageChunk(page),
+        )
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        var list = db
+
+        if (query.isNotBlank()) {
+            val stdQuery = query.replace(Regex("""\s+"""), "")
+            list = list.filter { it.name.contains(stdQuery) || it.author.contains(stdQuery) }
+        }
+
+        filters.filterIsInstance<ListFilter>().forEach {
+            list = it.applyFilter(list)
+        }
+
+        return Observable.just(
+            list.getPageChunk(page),
+        )
+    }
+
+    override fun getFilterList() = getFilters()
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> {
+        return client.newCall(mangaDetailsRequest(manga))
+            .asObservableSuccess()
+            .map { response ->
+                mangaDetailsParse(response, manga)
+            }
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        return GET("$baseUrl/azi_toon/${manga.url}.html", headers)
+    }
+
+    override fun getMangaUrl(manga: SManga): String {
+        return buildString {
+            if (currentBaseUrlHost.isBlank()) {
+                append(baseUrl)
+            } else {
+                append("https://")
+                append(currentBaseUrlHost)
+            }
+            append("/azi_toon/")
+            append(manga.url)
+            append(".html")
+        }
+    }
+
+    private fun mangaDetailsParse(response: Response, manga: SManga): SManga {
+        val document = response.asJsoup()
+        return manga.apply {
+            description = document.select("p.mt-2").last()?.text()
+            initialized = true
```

Comment:
Why not leave this in `fetchMangaDetails()`? Are you using `fetchMangaDetails()`  somewhere else where it makes sense that `initialized = false`?

If you remove the override for `fetchMangaDetails()` and this line it should be exactly the same.

Reply:
I initially override `fetchMangaDetails` to preserve the status, but now have passed it through the request so its fine